### PR TITLE
feat: Personal access token set up, replace gittar member token.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Personal access token set up, replace gittar member token.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTc4LDExNzQsMTE5MCwxMjE4LDEyMjRdLCJhc3NpZ25lZSI6WyIxMDAxMDczIl19&id=295887&iterationID=1190&pId=0&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Personal access token set up, replace gittar member token.           |
| 🇨🇳 中文    |    个人 token,  替换 gitta  token          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: conf/dop/dop.yaml [hunk 1]
```diff
@@ -75,6 +75,10 @@ grpc-client@erda.core.services.errorbox:
   addr: "${CORE_SERVICES_GRPC_ADDR:core-services:9537}"
 erda.core.services.errorbox-client: {}
 
+grpc-client@erda.core.token:
+  addr: "${CORE_SERVICES_GRPC_ADDR:core-services:9537}"
+erda.core.token-client:
```
Comment (Effet) [inline conf/dop/dop.yaml:80]: `erda.core.token-client: {}` is better

File: modules/core-services/providers/token/dao.go [hunk 2]
```diff
@@ -84,7 +91,10 @@ func (d *dao) QueryToken(ctx context.Context, req *pb.QueryTokensRequest) ([]tok
 }
 
 func (d *dao) CreateToken(ctx context.Context, req *pb.CreateTokenRequest) (*tokenstore.TokenStoreItem, error) {
-	obj := toModel(req)
+	obj, err := toModel(apis.GetUserID(ctx), req)
```
Comment (Effet) [inline modules/core-services/providers/token/dao.go:94]: Not good to directly get userId in `dao`, you should add Creator filed in `CreateTokenRequest `.

File: modules/core-services/services/member/member.go [hunk 3]
```diff
@@ -210,6 +219,17 @@ func (m *Member) CreateOrUpdate(userID string, req apistructs.MemberAddRequest)
 			// 创建成员
 			if len(members) == 0 {
 				orgID, projectID, applicationID := m.getIDs(req, targetScopeID)
+				if req.Scope.Type == apistructs.OrgScope {
```
Comment (Effet) [inline modules/core-services/services/member/member.go:222]: better add a TODO comment, we will not auto create PAT when user invited to org member, after we finish the page for PAT management.

File: modules/dop/services/project/project_package.go [hunk 4]
```diff
@@ -551,12 +556,20 @@ func (t *PackageDataDirector) makeGittarRemoteUrl() (string, error) {
 	}
 	logrus.Infof("gittar url %s", gittar)
 
-	members, err := t.bdl.GetMemberByUserAndScope(apistructs.OrgScope, t.Creator.GetContext().UserID, t.Creator.GetContext().OrgID)
+	res, err := t.tokenService.QueryTokens(context.Background(), &tokenpb.QueryTokensRequest{
+		Scope:     string(apistructs.OrgScope),
+		ScopeId:   strconv.FormatUint(t.Creator.GetContext().OrgID, 10),
+		Type:      mysqltokenstore.PAT.String(),
+		CreatorId: t.Creator.GetContext().UserID,
+	})
```
Comment (Effet) [inline modules/dop/services/project/project_package.go:564]: make a FIXME comment, this is very dangerous, project package may cause token leak.

File: modules/dop/endpoints/pipeline.go [hunk 5]
```diff
@@ -434,7 +441,7 @@ func (e *Endpoints) UpdateCmsNsConfigs(userID string, orgID uint64) error {
 			PipelineSource: apistructs.PipelineSourceDice.String(),
 			KVs: map[string]*cmspb.PipelineCmsConfigValue{
 				utils.MakeOrgGittarUsernamePipelineCmsNsConfig(): {Value: "git", EncryptInDB: true},
-				utils.MakeOrgGittarTokenPipelineCmsNsConfig():    {Value: members[0].Token, EncryptInDB: true}},
+				utils.MakeOrgGittarTokenPipelineCmsNsConfig():    {Value: res.Data[0].AccessKey, EncryptInDB: true}},
```
Comment (Effet) [inline modules/dop/endpoints/pipeline.go:444]: mark a TODO comment, we will use another way to help pipelines get good token, but that is not PAT.

**Review by Effet — CHANGES_REQUESTED** [review]

File: conf/dop/dop.yaml (hunk 1 above)
Comment (shuofan) [reply conf/dop/dop.yaml]: done

File: modules/core-services/providers/token/dao.go (hunk 2 above)
Comment (shuofan) [reply modules/core-services/providers/token/dao.go]: done

File: modules/core-services/services/member/member.go (hunk 3 above)
Comment (shuofan) [reply modules/core-services/services/member/member.go:224]: done

File: modules/dop/services/project/project_package.go (hunk 4 above)
Comment (shuofan) [reply modules/dop/services/project/project_package.go:564]: Added comment.

File: modules/dop/endpoints/pipeline.go (hunk 5 above)
Comment (shuofan) [reply modules/dop/endpoints/pipeline.go:444]: done

**Review by Effet — APPROVED** [review]

**Pull request merged**